implementations/rust/ockam/ockam_api/src/cli_state.rs

```diff
@@ -159,22 +155,30 @@ impl VaultsState {
         Ok(vaults)
     }
 
-    pub fn delete_by_state(&self, vault: &VaultState) -> Result<()> {
+    pub async fn delete(&self, name: &str) -> Result<()> {
+        // Retrieve vault. If doesn't exist do nothing.
+        let vault_state = match self.get(name) {
+            Ok(v) => v,
+            Err(CliStateError::NotFound(_)) => return Ok(()),
+            Err(e) => return Err(e),
+        };
+
+        // Set default to another vault if it's the default
```

Comment:
I think we should revisit this  (and even more the identity case https://github.com/build-trust/ockam/blob/d5bfa7ff85363378cdfc0791f1528c16e53f90db/implementations/rust/ockam/ockam_api/src/cli_state.rs#L348).

If users remove the identity that is pointed by as default,   then node will fail the next time it's tried to be run unless a new default is set.    That make sense and is expected,    while changing the default behind the scenes is not.
But we can do that in a separate PR

Reply:
I'll leave it as "resolved" so we can merge this PR. I'll take a look at this shortly, thanks for the review!